R/mod_downloadHandler.R

```diff
@@ -40,45 +40,101 @@ mod_downloadHandler_filetype_ui <- function(id){
+  uiOutput(NS(id, "mod_downloadHandler_incl_output"))
 }
-
-#' downloadHandler Inlcude Server Function
+#' downloadHandler Include Server Function
 #'
 #' @description A shiny Module.
 #'
 #' @param id Internal parameters for {shiny}.
+#' @param pkg_name the name of the package passed by mod_reportPreview
+#' @param user char vector containing name and role
+#' @param parent the parent (calling module) session information
+#' 
+#' @importFrom shiny showModal modalDialog
+#' @importFrom glue glue
+#' @importFrom shinyWidgets prettyCheckboxGroup updatePrettyCheckboxGroup
 #'
 #' @noRd 
-mod_downloadHandler_include_server <- function(id) {
+mod_downloadHandler_include_server <- function(id, pkg_name, user, parent) {
   moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+    
+    rpt_choices <- c("Report Author", "Report Date", "Risk Score", "Overall Comment", "Package Summary",
+                     "Maintenance Metrics", "Maintenance Comments", "Community Usage Metrics", "Community Usage Comments",
+                     "Source Explorer Comments")
+    
+    counter <- reactiveVal(value = 0L)
+    user_file <- reactiveVal(value = NULL)
+    
+    output$mod_downloadHandler_incl_output <- renderUI({
+      div(
+        strong(p("Elements to include:")),
+        div(align = 'left', class = 'twocol', style = 'margin-top: 0px;',
+            shinyWidgets::prettyCheckboxGroup(
+              ns("report_includes"), label = NULL, inline = FALSE,
+              choices = rpt_choices, selected = rpt_choices
+            )
+        ),
+        actionButton(ns("store_prefs"), "Store Preferences")
+      )
+    })
+    
     observe({
-      # print("#####################################")
-      # print("input$report_includes:")
-      # print(input$report_includes)
-      # shinyWidgets::updatePrettyCheckboxGroup("report_includes",)
+     req(counter() == 0L)
+     req(parent$input$tabs == "Build Report")
+     
+    # retrieve user data, if it exists.  Otherwise use rpt_choices above.
+     user_file(glue::glue("./inst/report_downloads/report_prefs_{user$name}_{user$role}.txt"))
+     if (file.exists(user_file())) {
+       session$userData$report_includes <- readLines(user_file())
+     } else {
+       session$userData$report_includes <- paste(rpt_choices, collapse = ",")  
+     }
+    counter(counter() + 1L)
 
-      # selected = ifelse(is.null(input$report_includes) | input$report_includes != my_choices,
-      #                   isolate(input$report_includes), my_choices)
+    }, priority = 2)
+      
+    # save user selections to session$userData$report_includes, and notify user
+    observeEvent(input$store_prefs, {
+     
+      session$userData$report_includes <- paste(input$report_includes, collapse = ",")
+      shiny::showModal(shiny::modalDialog(title = "User preferences saved",
+                                          footer = modalButton("Dismiss"), 
+                                          easyClose = TRUE))
```

Comment:
I don't really like how this modal looks with a title and footer but not body. Can we add something like "Report preferences stored for user" or akin to that?
![image](https://github.com/pharmaR/riskassessment/assets/85879620/cb91287f-e518-4ecf-8378-37b9c464c5da)

Reply:
done

Reply:
added "Report preferences stored for user" to body